pkg/utils/ocm.go

```diff
@@ -171,13 +178,52 @@ func GetHiveShard(clusterID string) (string, error) {
 		shard = shardPath.Body().HiveConfig().Server()
 	}
 
+	if out.Hypershift().Enabled() {
+		return getHiveURLforHCP(GetCurrentOCMEnv(connection)), nil
```

Comment:
Is this meant to get the hive cluster for a management cluster? If so, management clusters are just OSD/ROSA and the hive should be available in OCM.

If this is meant to  get the management cluster for a hypershift cluster, that should also be available in OCM. 

How to get Hive Shard from an OSD/ROSA cluster: 
https://github.com/openshift-online/ocm-cli/blob/main/pkg/cluster/describe.go#L109

How to get Management Cluster for a Hypershift HCP:
https://github.com/openshift-online/ocm-cli/blob/main/pkg/cluster/describe.go#L161-L170

Reply:
nice suggestion 👍 , I can try that
the reason I was leaning on hard-coding those was: https://gitlab.cee.redhat.com/service/backplane-cli/-/merge_requests/234/diffs#7fb99b6f8725a63d23914a8668dedf12d1872f30_109_135, I'm not sure if this is fixed though

Reply:
now that part is added as a check to replace the `clusterID` with corresponding management clustersID